packages/api/cms-api/src/file-uploads/file-uploads.service.ts

```diff
@@ -90,4 +91,14 @@ export class FileUploadsService {
 
         return ["/file-uploads", hash, file.id, timeout, resizeWidth, filename].join("/");
     }
+
+    async delete(fileUpload: FileUpload): Promise<void> {
+        const filePath = createHashedPath(fileUpload.contentHash);
+        if (await this.blobStorageBackendService.fileExists(this.config.directory, filePath)) {
+            await this.blobStorageBackendService.removeFile(this.config.directory, filePath);
+        }
+
+        this.entityManager.remove(fileUpload);
+        await this.entityManager.flush();
```

Comment:
usually a service should not flush as side effect, as this:
- will flush everything, not just the removed fileUpload
- will possibly flush too early, other operations might follow
- it's not clear to the user of this api that it flushes

Of course, this is kind of special as the file is removed and so it is very important to flush

Reply:
> Of course, this is kind of special as the file is removed and so it is very important to flush

Is it? The DAM also works if the file of an entity is missing.

It would be okay to remove the flush here or add an option `flush` so it's explicit in the application. But it's also okay to flush.

Reply:
I would suggest to remove the flush.

Reply:
I finally removed the flush: f4dd1aed610197088f45876fdec65bcc27ec1912